source/exe/signal_action.cc

```diff
@@ -42,21 +42,24 @@ void SignalAction::installSigHandlers() {
   stack.ss_size = altstack_size_;        // ... guard page at the other
   stack.ss_flags = 0;
 
-  RELEASE_ASSERT(sigaltstack(&stack, nullptr) == 0);
+  RELEASE_ASSERT(sigaltstack(&stack, &previous_altstack_) == 0);
 
+  int hidx = 0;
```

Comment:
Nit: I'd suggest it'd be clearer using a `std::list` or `std::vector` here instead of doing the array access index and size calculations.